src/string.cr

```diff
@@ -3015,6 +3112,20 @@ class String
     {bytes, bytesize}
   end
 
+  private def range_to_index_and_size(range)
+    from = range.begin
+    from += size if from < 0
+    raise IndexError.new if from < 0
```

Comment:
Include the `from` in the message, something like `#{from} out of bounds`?

To be honest, I'm not sure these error messages are very useful. If you get `IndexError` you have a bug in your program. If you see "3 not in 0..2" you don't get a lot of information to fix the bug just from that info. You'll have to debug it anyway to understand why that 3 got there in the first place.

But I can add a message. What message?